### ISSUE TITLE
When setting mode from readonly to design, tinymce steals focus

### ISSUE THREAD

**Issue by d3vkit** [tinymce/tinymce#4623]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

- When tinymce is in `readonly` mode
- If it is set to `design` mode
- It automatically sets focus to the text field

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tinymce.com](http://fiddle.tinymce.com/) or similar.**

Here is a demo: http://fiddle.tinymce.com/f2gaab/1

- Tinymce is set to readonly on load
- Click "toggle enabled" button to set tinymce to "design" mode

**What is the expected behavior?**

Tinymce should not take focus when set to design mode.

I have a form that has many fields disabled, one of which is a tinymce text area. When enabling the form, all fields become enabled, but tinymce is not the first field in the form, and it is unexpected that it would take focus.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**

Versions: I have tested this in edge and 4.8.3 in the fiddle.
Browser: Chrome
OS: Mac OSX

I don't know if this worked in previous versions.

**Comment by TheSpyder** [comment]
This is the intended behaviour, so what you'd need is a configuration option to not focus.

**Comment by kfowlks** [comment]
I require this functionality as well.

**Comment by glenfordwilliams** [comment]
is there a workaround to this?

**Comment by kfowlks** [comment]
Nope, not to my knowledge. We have a support contract and they advised us to upgrade to TinyMCE 5 the issue has been addressed but it was not back ported.

**Comment by gmqz** [comment]
`setMode(mode)` method (v4) triggers a focus event after changing from readonly to design mode.
Seems V5 doesn't have this unexpected behavior becose `setMode` method was deprecated and use `editor.mode.set(mode)` instead.

## Workaround v4
Instead use `setMode() `you can set the `contenteditable` property manually

> editor.getBody().setAttribute('contenteditable', true\false);

**Comment by Gouvernathor** [comment]
I'm on v3 (3.4.7 to be precise) and my code doesn't actually use setMode ; instead, it's calling `tinymce.init(config)`, and that one sets the focus to the element unless readonly is set to true in the config.
So, I tried to get the readonliness of the config, set it to true when passed to the init, and then convert the editor back to editable afterwards.
I tried the following :
```js
const editor = tinymce.get(config.elements);

editor.on('init', function () {
    editor.setMode("design");
});
```
When it works, the editor doesn't take focus when setMode gets called (for some reason ; I first tried to copy the code from setMode and remove the focus, turns out that's not even necessary). But half of the time it doesn't work and the editor remains readonly.
Is there a reliable way of doing this ?
